core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/HighlightField.java

```diff
@@ -112,4 +119,67 @@ public void writeTo(StreamOutput out) throws IOException {
             }
         }
     }
+
+    public static HighlightField fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        assert token == XContentParser.Token.START_OBJECT;
```

Comment:
I think assert makes sense because it is really a bug. I wonder if it makes sense to actually validate our own responses while parsing them. I think we should rely on tests for that, that our responses are correctly printed out using toXContent.